### PR TITLE
Feature/pausable butter

### PR DESCRIPTION
**Pull request by RedVeil**

### CODE REVIEW

File: packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol [hunk 1]
```diff
@@ -698,4 +702,13 @@ contract HysiBatchInteraction {
     IACLRegistry(contractRegistry.getContract(keccak256("ACLRegistry"))).requireRole(keccak256("DAO"), msg.sender);
     redeemThreshold = _threshold;
   }
+
+  /**
+   * @notice Pauses the contract.
+   * @dev All function with the modifer `whenNotPaused` cant be called anymore. Namly deposits and mint/redeem
+   */
+  function pause() external {
+    IACLRegistry(contractRegistry.getContract(keccak256("ACLRegistry"))).requireRole(keccak256("DAO"), msg.sender);
```
Comment (ecmendenhall) [inline packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol:711]: This pattern is all over the place now, and it's a little difficult to scan these lines for the important information (the authorized role). How about extracting these to a separate, inheritable `ACLAuth` contract? Something like:

```solidity
contract ACLAuth {
    IContractRegistry public contractRegistry;

    constructor(IContractRegistry _contractRegistry) {
        contractRegistry = _contractRegistry;
    }

    modifier requireRole(string memory role) {
          IACLRegistry(contractRegistry.getContract(keccak256("ACLRegistry"))).requireRole(keccak256(role), msg.sender);
          _;
    }

    // Similar modifiers + client functions for permissions, etc.
  
}

File: packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol [hunk 2]
```diff
@@ -698,4 +702,13 @@ contract HysiBatchInteraction {
     IACLRegistry(contractRegistry.getContract(keccak256("ACLRegistry"))).requireRole(keccak256("DAO"), msg.sender);
     redeemThreshold = _threshold;
```
Comment (ecmendenhall) [inline packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol:703]: Should we be emitting events on these state changes? None of these parameter setters (`setRedeemThreshold`, `setMintThreshold`, `setBatchCooldown`, `setCurveTokenPairs`) emit events.

File: packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol (hunk 1 above)
Comment (anthonymartin) [reply packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol:715]: @ecmendenhall good suggestion. I've created a ticket for this one

File: packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol (hunk 2 above)
Comment (anthonymartin) [reply packages/hardhat/contracts/core/defi/butter/ButterBatchProcessing.sol:707]: done

**Pull request merged**